src/rard/research/views/work.py

```diff
@@ -83,18 +88,48 @@ def get_success_url(self, *args, **kwargs):
 
     def form_valid(self, form):
         work = form.save(commit=False)
-        updated = form.cleaned_data["antiquarians"]
-        existing = work.antiquarian_set.all()
-        to_remove = [a for a in existing if a not in updated]
-        to_add = [a for a in updated if a not in existing]
-        for a in to_remove:
-            a.works.remove(work)
-        for a in to_add:
-            a.works.add(work)
-        add_new_books_to_work(work, form)
+        if "antiquarians" in form.cleaned_data:
+            updated = form.cleaned_data["antiquarians"]
+            existing = work.antiquarian_set.all()
+            to_remove = [a for a in existing if a not in updated]
+            to_add = [a for a in updated if a not in existing]
+            for a in to_remove:
+                a.works.remove(work)
+            for a in to_add:
+                a.works.add(work)
+            add_new_books_to_work(work, form)
         return super().form_valid(form)
 
 
+class WorkUpdateIntroductionView(WorkUpdateView):
+    model = Work
+    form_class = WorkIntroductionForm
+    permission_required = ("research.change_work",)
+    template_name = "research/work_detail.html"
+
+    def create_intro_if_does_not_exist(self, *args, **kwargs):
+        work = self.get_object()
+        if work.introduction is None:
+            work.introduction = TextObjectField.objects.create(content="")
+            work.save()
+
+    def dispatch(self, request, *args, **kwargs):
+        self.create_intro_if_does_not_exist()
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_success_url(self, *args, **kwargs):
```

Comment:
This is almost identical to the get_success_url function in the parent WorkUpdateView class. Is this necessary?